src/openforms/submissions/logic/rules.py

```diff
@@ -161,3 +162,60 @@ def iter_evaluate_rules(
                 else:
                     operation = compile_action_operation(action)
                     yield operation
+
+
+def get_targeted_components(
+    rule: FormLogic,
+    component_key_lookup: dict,
+    components_map: dict,
+    all_variables: Dict[str, FormVariable],
+    initial_data: dict,
+) -> List[dict]:
```

Comment:
I think I'd prefer this method to also invoke `compile_action_operation`, and have every (relevant class) implement the `get_targeted_components`. I want to avoid if/else mayhem based on the logic action type, but instead delegate that to a specialized class.